keras_hub/src/models/retinanet/retinanet_object_detector.py

```diff
@@ -18,6 +15,10 @@
     RetinaNetObjectDetectorPreprocessor,
 )
 
+# Check if Keras version is greater than or equal to 2.10.0
+if version.parse(keras.__version__) < version.parse("3.8.0"):
+    raise ImportError("Requires 3.8.0 or higher.")
+
 
 @keras_hub_export("keras_hub.models.RetinaNetObjectDetector")
 class RetinaNetObjectDetector(ImageObjectDetector):
```

Comment:
We need to align on names here. Either `ImageObjectDetector` and `RetinaNetImageObjectDetector`, or `ObjectDetector` and `RetinaNetObjectDetector`. That's our general pattern.

I'd prefer just rename `ImageObjectDetector` -> `ObjectDetector` everywhere. cc @divyashreepathihalli